sensu/plugins/check-disk-by-mount.rb

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env /opt/sensu/embedded/bin/ruby
+#
+# Check Disk Plugin
+# ===
+#
+# Uses GNU's -T option for listing filesystem type; unfortunately, this
+# is not portable to BSD. Warning/critical levels are percentages only.
+#
+# Copyright 2011 Sonian, Inc <chefs@sonian.net>
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
+
+require 'rubygems' if RUBY_VERSION < '1.9.0'
+require 'sensu-plugin/check/cli'
+
+class CheckDiskByMount < Sensu::Plugin::Check::CLI
+
+  option :mnt,
+    :short => '-m MNT',
+    :proc => proc {|a| a.split(',') }
+
+  option :warn,
+    :short => '-w PERCENT',
+    :proc => proc {|a| a.to_i },
+    :default => 85
+
+  option :crit,
+    :short => '-c PERCENT',
+    :proc => proc {|a| a.to_i },
+    :default => 95
+
+  def initialize
+    super
+    @crit_fs = []
+    @warn_fs = []
+  end
+
+  def read_df
+    `df -PT`.split("\n").drop(1).each do |line|
+      begin
+        _fs, type, _blocks, _used, _avail, capacity, mnt = line.split
+        next if config[:mnt] && !config[:mnt].include?(mnt)
+      rescue
+        unknown "malformed line from df: #{line}"
+      end
+      if capacity.to_i >= config[:crit]
+        @crit_fs << "#{mnt} #{capacity}"
+      elsif capacity.to_i >= config[:warn]
+        @warn_fs <<  "#{mnt} #{capacity}"
+      end
+    end
+  end
+
+  def usage_summary
+    (@crit_fs + @warn_fs).join(', ')
+  end
+
+  def run
+    read_df
+    critical usage_summary unless @crit_fs.empty?
+    warning usage_summary unless @warn_fs.empty?
```

Comment:
This should be updated to support criticality option, so that we can define per-environment whether or not a critical level would cause a critical alert (a page).